corehq/apps/api/odata/views.py

```diff
@@ -35,10 +39,13 @@ class ODataMetadataView(View):
     @method_decorator(api_auth)
     @method_decorator(toggles.ODATA.required_decorator())
     def get(self, request, domain):
-        # todo: should generate this dynamically based on the domain / case schema / data dictionary
         data_file = os.path.join(os.path.dirname(__file__), 'metadata.xml')
         with open(data_file, 'r') as f:
-            return add_odata_headers(HttpResponse(f.read(), content_type='application/xml'))
+            metadata_template = Template(f.read())
+        metadata = metadata_template.render(Context({
```

Comment:
Any reason you couldn't just use `render_to_string` here and not bother with doing the `Template`/`file` stuff manually?

Reply:
https://github.com/dimagi/commcare-hq/pull/23207/commits/7262b6773b471699acf2ecdbe1ec1a6bfbb58a9e